src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java

```diff
@@ -100,42 +128,42 @@ public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskLi
         Job job = run.getParent();
 
         Computer computer;
-        if (executor != null){
+        if (executor != null) {
             computer = executor.getOwner();
         } else if (Computer.currentComputer() != null) {
             computer = Computer.currentComputer();
-        } else if(fp.toComputer() != null){
+        } else if (fp.toComputer() != null) {
             computer = fp.toComputer();
         } else {
-            log("Can't hold as theres no node allocated :(");
+            log("Can't hold as there are no nodes allocated :(");
             return;
         }
 
-        if (computer instanceof NodePoolComputer){
+        if (computer instanceof NodePoolComputer) {
+            NodePoolComputer npc = (NodePoolComputer) computer;
+            NodePoolSlave nps = (NodePoolSlave) npc.getNode();
             try {
```

Comment:
Why move the try?

Reply:
So I could get access to the NPC object to print out the node name if an exception occurs.  Just printing a stack trace will work, but a meaningful error message with details is better.